streams/runners/storm/layout/src/main/java/com/hortonworks/streamline/streams/layout/storm/KafkaSpoutFluxComponent.java

```diff
@@ -86,178 +50,177 @@ private String addSpoutConfigComponent () {
+        String retryDelayPeriodMsId = "retryDelayPeriodMs" + UUID_FOR_COMPONENTS;
+        addTimeInterval(retryDelayPeriodMsId, conf.get("retryDelayPeriodMs") != null ? conf.get("retryDelayPeriodMs") : new Long(2));
+        constructorArgs.add(getRefYaml(retryDelayPeriodMsId));
+        constructorArgs.add(getConstructorArgsYaml(conf.get("maximumRetries") != null ? new Object[] {conf.get("maximumRetries")} : new Object[] {Integer
+                .MAX_VALUE}));
+        String retryDelayMaximumMs = "retryDelayMaximumMs" + UUID_FOR_COMPONENTS;
+        addTimeInterval(retryDelayMaximumMs, conf.get("retryDelayMaximumMs") != null ? conf.get("retryDelayMaximumMs") : new Long(10000));
+        constructorArgs.add(getRefYaml(retryDelayMaximumMs));
+        addToComponents(createComponent(componentId, className, null, constructorArgs, null));
+        return componentId;
     }
 
-    private void validateBooleanFields () throws ComponentConfigException {
-        String[] optionalBooleanFields = {
-            TopologyLayoutConstants.JSON_KEY_IGNORE_ZK_OFFSETS,
-            TopologyLayoutConstants.JSON_KEY_USE_START_OFFSET_IF_OFFSET_OUT_OF_RANGE
-        };
-        validateBooleanFields(optionalBooleanFields, false);
+    private void addTimeInterval (String componentId, Object value) {
+        String className = "org.apache.storm.kafka.spout.KafkaSpoutRetryExponentialBackoff$TimeInterval";
+        List<Object> constructorArgs = new ArrayList<>();
+        constructorArgs.add(value);
+        constructorArgs.add("MILLISECONDS");
+        addToComponents(createComponent(componentId, className, null, constructorArgs, null));
     }
 
-    private void validateStringFields () throws ComponentConfigException {
-        String[] requiredStringFields = {
-            TopologyLayoutConstants.JSON_KEY_ZK_URL,
-            TopologyLayoutConstants.JSON_KEY_TOPIC,
-            TopologyLayoutConstants.JSON_KEY_CONSUMER_GROUP_ID
-        };
-        validateStringFields(requiredStringFields, true);
-        String[] optionalStringFields = {
-            TopologyLayoutConstants.JSON_KEY_ZK_PATH
-        };
-        validateStringFields(optionalStringFields, false);
+    private String addRecordTranslator () {
+        String translatorId = "avroKafkaSpoutTranslator" + UUID_FOR_COMPONENTS;
+        String translatorClassname = "com.hortonworks.streamline.streams.runtime.storm.spout.AvroKafkaSpoutTranslator";
+        List<Object> constructorArgs = new ArrayList<>();
+        constructorArgs.addAll(getConstructorArgsYaml(new String[]{TopologyLayoutConstants.JSON_KEY_OUTPUT_STREAM_ID, TopologyLayoutConstants
+                .JSON_KEY_TOPIC}));
+        constructorArgs.add(kafkaSource != null ? kafkaSource.getId() : "");
+        constructorArgs.add(conf.get(TopologyLayoutConstants.SCHEMA_REGISTRY_URL));
+        constructorArgs.add(conf.get("readerSchemaVersion") != null ? Integer.parseInt((String) conf.get("readerSchemaVersion")) : null);
```

Comment:
@harshach found out this issue
agree that readerSchemaVersion integer value should be null for empty and null strings.